src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -103,12 +103,18 @@ private bool TrySetupConfigFileWatcher()
         {
             if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
             {
-                FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
-                _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
+                try
+                {
+                    FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
+                    _configFileWatcher = new(this, loader.GetConfigDirectoryName(), loader.GetConfigFileName());
+                    _hasAttemptedFileWatcherConfiguration = true;
+                    return true;
+                }
+                catch (Exception)
+                {
+                    return false;
+                }
             }
-
-            _hasAttemptedFileWatcherConfiguration = true;
-            return true;
         }
 
         return false;
```

Comment:
this line of code is unreachable now, instead of returning from both the try and catch blocks, you can simply return if `_configFileWatcher is not null` at the end